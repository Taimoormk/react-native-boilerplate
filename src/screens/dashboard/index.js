export { Dashboard } from './dashboard';
export { DashboardMobile } from './components/mobile';
export { DashboardWeb } from './components/web';
