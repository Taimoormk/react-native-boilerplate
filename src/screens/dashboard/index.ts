export * from './dashboard';
export * from './mobile';
export * from './web';
