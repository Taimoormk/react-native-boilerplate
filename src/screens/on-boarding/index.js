export { OnBoarding } from './on-boarding';
