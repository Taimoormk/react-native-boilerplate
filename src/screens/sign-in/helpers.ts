export const loginScreenText : ILoginScreenText = {
  oAuthHeading: 'Continue with Facebook or Google',
  facebookButton: 'Continue with Facebook',
  googleButton: 'Continue with Google',
  signinHeading: 'Enter Credentials to begin',
};

export interface ILoginScreenText {
  [key : string] : string;
}
