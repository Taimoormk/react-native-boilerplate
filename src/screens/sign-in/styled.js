// ########## Import Dependencies Here ##########
import { Text, View } from '@react-pakistan/react-native-commons-collection';
import styled from 'styled-components/native';

export const SignInWrapper = styled(View)`
  align-items: center;
  background-color: #CCCCCC;
  flex: 1;
  justify-content: center;
`;

export const OAuthHeading = styled(Text)`
  color: ${({ theme }) => theme.colors.black};
  font-family: ${({ theme }) => theme.typography.fontFamily};
  font-size: ${({ theme }) => theme.spacing.default}px;
  text-transform: uppercase;
`;

export const Spacer = styled(View)`
  flex: 0;
  margin-vertical: ${({ theme }) => theme.spacing.small}px;
`;
