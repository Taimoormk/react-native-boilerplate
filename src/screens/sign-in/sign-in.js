// ########## Import Dependencies Here ##########
import { Button } from '@react-pakistan/react-native-commons-collection';
import { Spacer } from '@react-pakistan/react-native-commons-collection/spacer';
import { theme as defaultTheme } from '@react-pakistan/util-react-native-functions';
import { object } from 'prop-types';
import React from 'react';
import { useDispatch } from 'react-redux';
import { withTheme } from 'styled-components';
import {
  OAuthHeading,
  SignInWrapper,
} from './styled';

// ########## Import Components Here ##########
import { loginScreenText } from './helpers';
import { showAppAction } from '../../redux/actions';

export const SignIn = withTheme(({
  theme,
}) => {
  // dispatch
  const dispatch = useDispatch();

  return (
    <SignInWrapper>
      <OAuthHeading>
        {loginScreenText.oAuthHeading}
      </OAuthHeading>
      <Spacer margin={theme.spacing.default} />
      <Button
        onPress={() => dispatch(showAppAction())}
        title={loginScreenText.facebookButton}
      />
      <Spacer margin={theme.spacing.default} />
      <Button
        onPress={() => dispatch(showAppAction())}
        title={loginScreenText.googleButton}
      />
    </SignInWrapper>
  );
});

SignIn.propTypes = {
  theme: object,
};
SignIn.defaultProps = {
  theme: defaultTheme,
};
