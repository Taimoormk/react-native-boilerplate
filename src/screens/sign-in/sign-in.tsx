import { Button, Spacer } from '@react-pakistan/react-native-commons-collection';
import { IRNTheme } from '@react-pakistan/util-react-native-functions';
import React, { ReactElement, memo, useCallback } from 'react';
import { useDispatch } from 'react-redux';
import { withTheme } from 'styled-components/native';
import { showAppAction } from '../../redux/actions';
import { loginScreenText } from './helpers';
import {
  OAuthHeading,
  SignInWrapper,
} from './styled';

export const SignIn = memo(withTheme(({
  theme,
} : ISignInProps) : ReactElement => {
  // dispatch
  const dispatch = useDispatch();

  // callbacks
  const onPressHandler = useCallback(() : void => {
    dispatch(showAppAction());
  }, []);

  return (
    <SignInWrapper>
      <OAuthHeading>
        {loginScreenText.oAuthHeading}
      </OAuthHeading>
      <Spacer marginVertical={theme.spacing.xsmall} />
      <Button
        onPress={onPressHandler}
        title={loginScreenText.facebookButton}
      />
      <Spacer marginVertical={theme.spacing.xsmall} />
      <Button
        onPress={onPressHandler}
        title={loginScreenText.googleButton}
      />
    </SignInWrapper>
  );
}));

export interface ISignInProps {
  /**
   *
   */
  theme : IRNTheme;
}
