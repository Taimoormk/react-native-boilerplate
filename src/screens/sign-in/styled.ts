import { Text, View } from '@react-pakistan/react-native-commons-collection';
import styled from 'styled-components/native';

export const SignInWrapper = styled(View)`
  align-items: center;
  background-color: ${({ theme }) => theme.colors.lightGrey};
  flex: 1;
  justify-content: center;
`;

export const OAuthHeading = styled(Text)`
  color: ${({ theme }) => theme.colors.black};
  font-family: ${({ theme }) => theme.typography.fontFamily};
  font-size: ${({ theme }) => theme.spacing.default}px;
  text-transform: uppercase;
`;
