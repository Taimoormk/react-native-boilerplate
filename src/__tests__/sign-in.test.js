// ########## Import Dependencies Here ##########
import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

// ########## Import Components Here ##########
import { SignIn } from '../screens/sign-in';

// ########## Create Initial State ##########
const fakeInitialState = {
  showAppReducer: {
    showApp: true,
  },
};

const mockStore = configureStore();

test('renders SignIn component correctly', () => {
  const renderer = new ShallowRenderer();
  const showApp = false;
  const store = mockStore(fakeInitialState);
  const showAppAction = () => null;
  renderer.render(
    <Provider store={store}>
      <SignIn showApp={showApp} showAppAction={showAppAction} />
    </Provider>,
  );
  const wrapper = renderer.getRenderOutput();
  expect(wrapper).toMatchSnapshot();
});
