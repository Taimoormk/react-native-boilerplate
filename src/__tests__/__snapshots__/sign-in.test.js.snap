// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders SignIn component correctly 1`] = `
<Context.Provider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <Memo(WithTheme(Component))
    showApp={false}
    showAppAction={[Function]}
    theme={
      Object {
        "breakpoints": Object {
          "desktop": "(min-width: 2560px)",
          "laptop": "(min-width: 1024px)",
          "laptopL": "(min-width: 1440px)",
          "mobileL": "(min-width: 425px)",
          "mobileM": "(min-width: 375px)",
          "mobileS": "(min-width: 320px)",
          "tablet": "(min-width: 768px)",
        },
        "colors": Object {
          "black": "#2E2E2E",
          "cherryRed": "#F62F5E",
          "darkGrey": "#EFEFEF",
          "inputBorder": "#E1E1E1",
          "lightBlack": "#6C6C6C",
          "lightGrey": "#F7F7F7",
          "lighterBlack": "#B4B4B4",
          "limeYellow": "#EFFC90",
          "paleGreen": "#00D3CA",
          "rustyGold": "#F1AD3D",
          "skyBlue": "#1E90FF",
          "white": "#FFFFFF",
        },
        "misc": Object {},
        "radius": Object {
          "default": 8,
          "double": 16,
          "full": 100,
          "half": 60,
        },
        "shadow": Object {
          "default": "0px 1px 4px #E5E9F2",
          "medium": "-4px 0 12px rgba(0, 0, 0, 0.3)",
        },
        "spacing": Object {
          "default": 32,
          "huge": 96,
          "large": 48,
          "medium": 24,
          "small": 16,
          "xlarge": 64,
          "xsmall": 8,
        },
        "typography": Object {
          "fontFamily": "Roboto-Regular",
          "h1": Object {
            "fontFamily": "Roboto-Bold",
            "fontSize": 56,
            "letterSpacing": 0,
            "lineHeight": 56,
            "textAlign": "left",
            "textTransform": "uppercase",
          },
          "h2": Object {
            "fontFamily": "Roboto-Medium",
            "fontSize": 44,
            "letterSpacing": 0,
            "lineHeight": 56,
            "textAlign": "left",
            "textTransform": "uppercase",
          },
          "h3": Object {
            "fontFamily": "Roboto-Regular",
            "fontSize": 32,
            "letterSpacing": 0,
            "lineHeight": 56,
            "textAlign": "left",
            "textTransform": "uppercase",
          },
          "text": Object {
            "fontFamily": "Roboto-Regular",
            "fontSize": 24,
            "letterSpacing": 0.25,
            "lineHeight": 36,
            "textAlign": "left",
            "textTransform": "none",
          },
        },
        "zIndex": Object {
          "dropdown": 100,
          "modal": 300,
          "pageFooter": 200,
        },
      }
    }
  />
</Context.Provider>
`;
